### PR TITLE
Fix `UMFPACK` to use `check=false`

### PR DESCRIPTION
**Pull request by oscardssmith**
This prevents singular exceptions.

## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API

### CODE REVIEW

File: test/default_algs.jl [hunk 1]
```diff
@@ -35,6 +35,13 @@ solve(prob)
     LinearSolve.OperatorAssumptions(false)).alg ===
       LinearSolve.DefaultAlgorithmChoice.QRFactorization
 
+
+A = spzeros(100, 100)
+A[1,1]=1
+# test that solving a singular problem doesn't error
+prob = LinearProblem(A, ones(100))
+solve(prob)
```
Comment (ChrisRackauckas) [inline test/default_algs.jl:43]: This only shows one solver. Can you make KLU and UMFPACK explicit?

Comment (oscardssmith) [reply test/default_algs.jl]: updated. KLU is still broken (as far as I can see it still doesn't support disabling the checks).

Comment (ChrisRackauckas) [reply test/default_algs.jl]: @rayegun didn't you handle KLU.jl?

Comment (ChrisRackauckas) [reply test/default_algs.jl]: https://github.com/SciML/LinearSolve.jl/pull/456#issuecomment-1935228441

https://github.com/search?q=repo%3AJuliaSparse%2FKLU.jl%20check&type=code

You said you added `check=false` to klu but I don't see it in the repo. We can merge for now with this test failing but can you follow up with a test passing verison?

**Pull request merged**